tests/testthat/test-export_cdisc.R

```diff
@@ -0,0 +1,309 @@
+        PCTPTNUM = rep(c(0, 1), each = 5),
+        PCTPTREF = "Most Recent Dose",
+        AVISIT = c("Visit 1", "Visit 2", "Visit 3", "Visit 4", "Visit 5",
+                   "Visit 1", "Visit 2", "Visit 3", "Visit 4", "Visit 5"),
+        VISIT = c("Visit 1", "Visit 2", "Visit 3", "Visit 4", "Visit 5",
+                  "Visit 1", "Visit 2", "Visit 3", "Visit 4", "Visit 5"),
+        AVISITN = c(1, 2, 3, 4, 5, 1, 2, 3, 4, 5),
+        VISITNUM = c(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)
+      ),
+      columns = list(
+        groups = c("USUBJID", "PARAM")
+      )
+    )
+  )
+)
+
+describe("export_cdisc", {
+  it("exports CDISC-compliant datasets (PP, ADPP, ADPC)", {
+    result <- export_cdisc(test_pknca_res)
+
+    # Check that the result contains the expected components
+    expect_type(result, "list")
+    expect_named(result, c("pp", "adpp", "adpc", "studyid"))
+
+    # Check the PP
+    pp <- result$pp
+    expect_s3_class(pp, "data.frame")
+    expect_true(all(CDISC_COLS$PP %in% names(pp)))
+    expect_equal(nrow(pp), 6) # Ensure all variables are included
+
+    # Check the ADPP
+    adpp <- result$adpp
+    expect_s3_class(adpp, "data.frame")
+    expect_true(all(CDISC_COLS$ADPP %in% names(adpp)))
+    expect_equal(nrow(adpp), 6) # Ensure all variables are included
+
+    # Check the ADPC
+    adpc <- result$adpc
+    expect_s3_class(adpc, "data.frame")
+    expect_true(all(CDISC_COLS$ADPC %in% names(adpc)))
+    expect_equal(nrow(adpc), 10) # Ensure all concentration data is included
+
+    # Check the studyid
+    expect_equal(result$studyid, "S1")
+  })
+
+  it("derives ATPT and ATPTN when they are not present", {
+    modified_test_pknca_res <- test_pknca_res
+    modified_test_pknca_res$data$conc$data <- modified_test_pknca_res$data$conc$data %>%
+      select(-ATPT, -ATPTN)
+
+    result <- export_cdisc(modified_test_pknca_res)
+
+    # Check that ATPT and ATPTN are derived as NA
+    expect_true("ATPT" %in% names(result$adpc))
+    expect_true("ATPTN" %in% names(result$adpc))
+    expect_true(all(!is.na(result$adpc$ATPT)))
+    expect_true(all(!is.na(result$adpc$ATPTN)))
+  })
+
+  it("derives PPSTINT and PPENINT for partial AUC intervals", {
+    modified_test_pknca_res <- test_pknca_res
+    modified_test_pknca_res$result <- modified_test_pknca_res$result %>%
+      mutate(PPTESTCD = ifelse(PPTESTCD == "AUCINT", "AUCINT", PPTESTCD))
+
+    result <- export_cdisc(modified_test_pknca_res)
+
+    # Check that PPSTINT and PPENINT are derived correctly
+    expect_true("PPSTINT" %in% names(result$pp))
+    expect_true("PPENINT" %in% names(result$pp))
+    expect_equal(result$pp$PPSTINT[which(result$pp$PPTESTCD == "AUCINT")], "PT2H")
+    expect_equal(result$pp$PPENINT[which(result$pp$PPTESTCD == "AUCINT")], "PT4H")
+  })
+
+  it("derives PPREASND & PPSTAT correctly in all situations", {
+    modified_test_pknca_res <- test_pknca_res
+    modified_test_pknca_res$result <- test_pknca_res$result %>%
```

Comment:
nitpick: Might be a little confusing why we are using the _old_ variable rather than just created localized one, it is basically the same, but this would be in my opinion a bit clearer:
```suggestion
    modified_test_pknca_res$result <- modified_test_pknca_res$result %>%
```